Tella/Scenes/Settings/ViewModel/GDriveServerViewModel.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  GDriveServerViewModel.swift
+//  Tella
+//
+//  Created by gus valbuena on 5/22/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import GoogleSignIn
+import GoogleAPIClientForREST
+
+class GDriveServerViewModel: ObservableObject {
+    var googleUser:GIDGoogleUser? = nil
+    @Published var sharedDrives: [GTLRDrive_Drive] = []
+    
+    init() {
+        GIDSignIn.sharedInstance.restorePreviousSignIn { [self] user, error in
```

Comment:
In my opininion, it's better to centralise the google drive functions in a repository and add a call from the view model to avoid redundancy and minimise the changes in case there is an update in the future. What do you think ?

Reply:
sure! I was thinking of doing something similar to centralize the g-drive functions. In this case we could create the repository for all the g-drive endpoint calls and unify both view models that I created right? so the GDriveServerVM and GDriveAuthVM will be just one

Reply:
` In this case we could create the repository for all the g-drive endpoint calls` :  yes 
but we can keep the GDriveServerVM and GDriveAuthVM to make the calls between views and the g-drive repository. What do you think ?

Reply:
@dhekra-rouatbi I created GDriveRepository for centralizing all the API calls and interactions with GDriveAPI. Since the GDriveAPI works different than the API we interact (tella web and uwazi) I didn't follow the pattern of those repository so I created a singleton instead. Let me know what do you think